src/battery-check/battery-check.c

```diff
@@ -0,0 +1,24 @@
+/* SPDX-License-Identifier: LGPL-2.1-or-later */
+
+#include "log.h"
+#include "main-func.h"
+#include "sleep-config.h"
+
+static int run(int argc, char *argv[]) {
+    int r;
+
+    log_parse_environment();
+    log_open();
+
+    r = battery_is_discharging_and_low();
+    if (r < 0) {
+        log_error_errno(r, "Failed to check battery status: %m");
+        return 0;
```

Comment:
> you are eating up the error here. we generally have to rule that we should log at LOG_WARNING level at most in such cases. also suffix with ", ignoring." in that case, so that it's clear to anyone reading this that the issue is ignored.
> 
> i.e.:
> 
> ```c
> if (r < 0) {
>         log_warning_errno(r, "Failed to check battery status, ignoring: %m");
>         return 0;
> }
> ```

Awesome..This is exciting.